File: hackerrank/warmup/compare-the-triplets.hs
-- Submission url : https://www.hackerrank.com/challenges/compare-the-triplets/submissions/code/168512207
solve :: [Int] -> [Int]
solve xs = let
    as = take 3 xs
    bs = drop 3 xs
    comp_list = [a `compare` b| (a,b) <- zip as bs]
    a_score = length (filter (== GT) comp_list)
    b_score = length (filter (== LT) comp_list)
    in [a_score,b_score]



main = interact $ unwords . map show . solve . map read . words

File: hackerrank/warmup/compare-the-triplets.py
# Submission url : https://www.hackerrank.com/challenges/compare-the-triplets/submissions/code/168586003
def solve(alst,blst):
    a, b = 0,0
    for a_i, b_i in zip(alst,blst):
        if a_i  > b_i:
            a += 1
        elif b_i > a_i:
            b += 1
    
    return [a,b]

if __name__ == '__main__':
    alst = map(int,input().split(" "))
    blst = map(int,input().split(" "))

    result = solve(alst,blst)
    result_str = map(str,result)
    print(" ".join(result_str))

